File: Data Structures & Algorithms/search-2d-matrix/submission-2.java
class Solution {
    public boolean bsearch(int[] arr, int low, int high, int target)
    {
        int mid ;
        while(low<=high)
        {
            mid= (low+high)/2;
            if(arr[mid]==target)
            {
                return true;
            }
            else if(arr[mid]>target)
            {
                low = mid+1;
            }
            else
            {
                high = mid -1;
            }
        }
        return false;
    }
    public boolean searchMatrix(int[][] matrix, int target) {
        int row;
        for (int i=0;i<matrix.length;i++)
        {
            if((target>=matrix[i][0])&&(target<=matrix[i][matrix[i].length-1]))
            {
                return bsearch(matrix[i],0,matrix[i].length-1,target);
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        start = 0
        end = len(numbers)-1

        while(start<end):
            if (target == numbers[start]+numbers[end]):
                return [start+1,end+1]
            elif target< numbers[start]+numbers[end]:
                end = end-1
            else:
                start = start+1

        return []
        